src/snowflake/cli/plugins/object/commands.py

```diff
@@ -32,6 +36,24 @@
     "all functions that begin with “my”.",
 )
 
+
+def _scope_callback(scope: Tuple[str, str]):
+    if scope[1] is not None and not is_valid_identifier(scope[1]):
+        raise ClickException("scope name must be a valid identifier")
+    if scope[0] is not None and scope[0].lower() not in VALID_SCOPES:
```

Comment:
If the object type is not service, --in compute-pool scope will not be supported. Do we have this validation?

Reply:
good point, my initial thought was that if they passed in a compute pool scope with something other than service they would get an SQL compilation error anyways but you're right that we should do as much frontend validation as possible.